public/app/features/provisioning/RepositoryListPage.tsx

```diff
@@ -26,18 +27,27 @@ import { useRepositoryList } from './hooks';
 export default function RepositoryListPage() {
   const [items, isLoading] = useRepositoryList({ watch: true });
   const settings = useGetFrontendSettingsQuery();
-  const navigate = useNavigate();
 
-  useEffect(() => {
-    if (settings.data?.legacyStorage && !items?.length) {
-      navigate('/admin/provisioning/setup');
-    }
-  }, [settings, navigate, items?.length]);
```

Comment:
^^ since items and settings come in at different times, this would navigate to the setup guide even when a repository actually exists (but is still loading)